deploy/pre-deployment/pre-deployment-check.sh

```
@@ -122,54 +125,101 @@ check_default_storage_class() {
+    local allocatable_gpu_nodes
+    allocatable_gpu_nodes=$(kubectl get nodes \
+        -o jsonpath='{range .items[*]}{.metadata.name}{" "}{.status.allocatable.nvidia\.com/gpu}{"\n"}{end}' \
+        2>/dev/null || true)
+
+    local allocatable_count
+    allocatable_count=$(echo "$allocatable_gpu_nodes" | awk '$2 != "" && $2 != "0"' | wc -l | tr -d ' ')
+
+    if [[ $allocatable_count -gt 0 ]]; then
+        MANAGED_GPU_DETECTED=true
+
+        print_status $GREEN "✅ Found ${allocatable_count} GPU node(s) via allocatable GPU resources"
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)